The Persistent Volume Claims page is not working

On the latest version (0.20.1) the PVC page is blank and headlamp has to be restarted.

On the console I can see the the following error.

```
apiProxy.ts:955 WebSocket connection to
'ws://localhost:4466/clusters/admin@talos-cluster/api/v1/persistentvolumeclaims?watch=1&resourceVersion=60924699' failed: WebSocket is closed before the connection is established.

apiProxy.ts:985 Error in api stream 
{err: Event, path: 'api/v1/persistentvolumeclaims?watch=1&resourceVersion=60924699'}
err: Event {isTrusted: true, type: 'error', target: WebSocket, currentTarget: WebSocket, eventPhase: 0, …}
path: "api/v1/persistentvolumeclaims?watch=1&resourceVersion=60924699"
```

But I could not see any error on the backend side.

--- a/frontend/src/components/storage/ClaimList.stories.tsx
+++ b/frontend/src/components/storage/ClaimList.stories.tsx
@@ -6,7 +6,7 @@ import PersistentVolumeClaim, {
 import { TestContext } from '../../test';
 import PVClaimList from './ClaimList';
 
-const basePVC = {
+const basePVC: KubePersistentVolumeClaim = {
   apiVersion: 'v1',
   kind: 'PersistentVolumeClaim',
   metadata: {
@@ -40,9 +40,21 @@ const basePVC = {
 PersistentVolumeClaim.useList = () => {
   const noStorageClassNamePVC = _.cloneDeep(basePVC);
   noStorageClassNamePVC.metadata.name = 'no-storage-class-name-pvc';
-  noStorageClassNamePVC.spec.storageClassName = '';
+  noStorageClassNamePVC.spec!.storageClassName = '';
 
-  const objList = [basePVC, noStorageClassNamePVC].map(
+  const noVolumeNamePVC = _.cloneDeep(basePVC);
+  noVolumeNamePVC.metadata.name = 'no-volume-name-pvc';
+  noVolumeNamePVC.spec = {
+    accessModes: ['ReadWriteOnce'],
+    volumeMode: 'Block',
+    resources: {
+      requests: {
+        storage: '10Gi',
+      },
+    },
+  };
+
+  const objList = [basePVC, noStorageClassNamePVC, noVolumeNamePVC].map(
     pvc => new PersistentVolumeClaim(pvc as KubePersistentVolumeClaim)
   );
   return [objList, null, () => {}, () => {}] as any;

--- a/frontend/src/components/storage/ClaimList.tsx
+++ b/frontend/src/components/storage/ClaimList.tsx
@@ -42,6 +42,9 @@ export default function VolumeClaimList() {
           label: t('Volume'),
           getter: volumeClaim => {
             const name = volumeClaim.spec.volumeName;
+            if (!name) {
+              return '';
+            }
             return (
               <Link routeName="persistentVolume" params={{ name }} tooltip>
                 {name}

--- a/frontend/src/lib/k8s/persistentVolumeClaim.ts
+++ b/frontend/src/lib/k8s/persistentVolumeClaim.ts
@@ -2,25 +2,27 @@ import { apiFactoryWithNamespace } from './apiProxy';
 import { KubeObjectInterface, makeKubeObject } from './cluster';
 
 export interface KubePersistentVolumeClaim extends KubeObjectInterface {
-  spec: {
-    accessModes: string[];
-    resources: {
+  spec?: {
+    accessModes?: string[];
+    resources?: {
       limits?: object;
       requests: {
         storage?: string;
         [other: string]: any;
       };
     };
-    storageClassName: string;
-    volumeMode: string;
-    volumeName: string;
+    storageClassName?: string;
+    volumeMode?: string;
+    volumeName?: string;
     [other: string]: any;
   };
-  status: {
+  status?: {
     capacity?: {
       storage?: string;
     };
-    phase: string;
+    phase?: string;
+    accessModes?: string[];
+    [other: string]: any;
   };
 }
 

